### PR TITLE
Align config logic with changes in python SDK

### PR DESCRIPTION
**Pull request by HonzaHrazdira**
##### SUMMARY

1. Newest version of Python SDK takes care of configuration and env vars parsing and therefore this logic can be removed from ansible modules. 
2. Allows fusion profile selection by setting config_profile argument in any fusion module.
##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

`module_utils.get_fusion()`

##### ADDITIONAL INFORMATION
I am creating a draft PR. Once the newest version of Python SDK is released we can merge it.

### CODE REVIEW

File: plugins/module_utils/fusion.py [hunk 1]
```diff
@@ -112,10 +111,6 @@ def get_fusion(module):
         module.fail_on_missing_params([PARAM_PRIVATE_KEY_FILE])
 
     config = fusion.Configuration()
-    if ENV_API_HOST in environ or ENV_HOST in environ:
-        host_url = environ.get(ENV_API_HOST, environ.get(ENV_HOST))
-        config.host = urljoin(host_url, BASE_PATH)
-    config.token_endpoint = environ.get(ENV_TOKEN_ENDPOINT, config.token_endpoint)
 
     if access_token is not None:
```
Comment (dimleevik) [inline plugins/module_utils/fusion.py:115]: I think you can even remove these if statements for `access_token`, `private_key_password`, `private_key_file` and just put all of them into fusion.Configuration() initialization

**Review by andreepyro — COMMENTED** [review]
Don't forget to update the `purefusion` version requirements as this changes uses features added only in specific version.

File: plugins/module_utils/fusion.py (hunk 1 above)
Comment (HonzaHrazdira) [reply plugins/module_utils/fusion.py]: Done in [88d10ac](https://github.com/Pure-Storage-Ansible/Fusion-Collection/pull/140/commits/88d10acf05094bfb848ab8ade89095d6ec09d52c)

File: changelogs/141_add_config_profile.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - all modules - added `config_profile` argument, which can be used instead of `user` and `principal` argument
```
[the remark this answers is not in the document]
Comment (HonzaHrazdira) [reply changelogs/141_add_config_profile.yml:2]: ouch. That was copy pasta.

**Comment by HonzaHrazdira** [comment]
Converting to draft again since the latest release does not contain the configuration update yet.

**Pull request closed without merging**